File: 94-BinaryTreeInorderTraversal/94-BinaryTreeInorderTraversal.java
// Last updated: 5/25/2026, 11:20:31 AM
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
      ArrayList<Integer> l=new ArrayList<>();
    public List<Integer> inorderTraversal(TreeNode root) {
        if(root!=null){
            inorderTraversal(root.left);
            l.add(root.val);
            inorderTraversal(root.right);
        }
        return l;
    }
}

File: 94-BinaryTreeInorderTraversal/94-BinaryTreeInorderTraversal.py
# Last updated: 5/25/2026, 11:20:36 AM
# Definition for a binary tree node.
# class TreeNode(object):
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution(object):
    def inorderTraversal(self, root):
        stack=[]
        result=[]
        current=root
        while current or stack:
            while current:
                stack.append(current)
                current=current.left
            current=stack.pop()
            result.append(current.val)
            current=current.right
        return result
        """
        :type root: Optional[TreeNode]
        :rtype: List[int]
        """
        